File: frontend/src/api/feedback.ts
import { get, post, put } from './request'
import type { Feedback, FeedbackCreate, FeedbackUpdate, PaginatedResponse } from '@/types'

// 获取反馈列表
export async function getFeedbacks(params?: {
  page?: number
  per_page?: number
  status?: string
  category?: string
  priority?: string
  search?: string
}): Promise<PaginatedResponse<Feedback>> {
  const queryParams = new URLSearchParams()
  
  if (params?.page) queryParams.append('page', params.page.toString())
  if (params?.per_page) queryParams.append('per_page', params.per_page.toString())
  if (params?.status) queryParams.append('status', params.status)
  if (params?.category) queryParams.append('category', params.category)
  if (params?.priority) queryParams.append('priority', params.priority)
  if (params?.search) queryParams.append('search', params.search)
  
  const response = await get<PaginatedResponse<Feedback>>(`/api/feedbacks?${queryParams.toString()}`)
  
  if (response.success && response.data) {
    return response.data
  }
  
  return {
    items: [],
    pagination: {
      page: 1,
      per_page: 10,
      total: 0,
      total_pages: 0
    }
  }
}

// 获取单个反馈
export async function getFeedback(id: number): Promise<Feedback | null> {
  const response = await get<Feedback>(`/api/feedbacks/${id}`)
  return response.success ? response.data || null : null
}

// 创建反馈
export async function createFeedback(data: FeedbackCreate): Promise<Feedback | null> {
  const response = await post<Feedback>('/api/feedbacks', data)
  return response.success ? response.data || null : null
}

// 更新反馈
export async function updateFeedback(id: number, data: FeedbackUpdate): Promise<Feedback | null> {
  const response = await put<Feedback>(`/api/feedbacks/${id}`, data)
  return response.success ? response.data || null : null
}

// 删除反馈
export async function deleteFeedback(id: number): Promise<boolean> {
  const response = await get<{ message: string }>(`/api/feedbacks/${id}/delete`)
  return response.success
}

// 获取反馈统计
export async function getFeedbackStats(): Promise<{
  total: number
  pending: number
  processing: number
  resolved: number
  closed: number
}> {
  const response = await get<{
    total: number
    pending: number
    processing: number
    resolved: number
    closed: number
  }>('/api/feedbacks/stats')
  
  if (response.success && response.data) {
    return response.data
  }
  
  return {
    total: 0,
    pending: 0,
    processing: 0,
    resolved: 0,
    closed: 0
  }
}

File: routes/feedback.py
from fastapi import APIRouter, Depends, HTTPException, Query
from sqlalchemy.orm import Session
from database import get_db, Feedback
from pydantic import BaseModel
from datetime import datetime
from typing import Optional, List

router = APIRouter()

class FeedbackCreate(BaseModel):
    content: str
    type: str
    rating: Optional[int] = None
    sentiment: Optional[str] = None
    sentiment_score: Optional[float] = None
    player_name: Optional[str] = None
    player_id: Optional[str] = None

class FeedbackUpdate(BaseModel):
    status: Optional[str] = None
    priority: Optional[str] = None

@router.post("/")
async def create_feedback(feedback: FeedbackCreate, db: Session = Depends(get_db)):
    try:
        db_feedback = Feedback(
            content=feedback.content,
            type=feedback.type,
            rating=feedback.rating,
            sentiment=feedback.sentiment,
            sentiment_score=feedback.sentiment_score,
            player_name=feedback.player_name,
            player_id=feedback.player_id
        )
        db.add(db_feedback)
        db.commit()
        db.refresh(db_feedback)
        return {"id": db_feedback.id, "message": "反馈提交成功"}
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

@router.get("/")
async def get_feedback(
    page: int = Query(1, ge=1),
    page_size: int = Query(20, ge=1, le=100),
    type: Optional[str] = None,
    status: Optional[str] = None,
    priority: Optional[str] = None,
    search: Optional[str] = None,
    db: Session = Depends(get_db)
):
    try:
        query = db.query(Feedback)
        
        if type:
            query = query.filter(Feedback.type == type)
        if status:
            query = query.filter(Feedback.status == status)
        if priority:
            query = query.filter(Feedback.priority == priority)
        if search:
            query = query.filter(Feedback.content.contains(search))
        
        total = query.count()
        feedbacks = query.order_by(Feedback.created_at.desc()).offset((page-1)*page_size).limit(page_size).all()
        
        return {
            "data": [
                {
                    "id": f.id,
                    "content": f.content,
                    "type": f.type,
                    "rating": f.rating,
                    "sentiment": f.sentiment,
                    "sentiment_score": f.sentiment_score,
                    "status": f.status,
                    "priority": f.priority,
                    "player_name": f.player_name,
                    "created_at": f.created_at.isoformat()
                } for f in feedbacks
            ],
            "total": total,
            "page": page,
            "page_size": page_size
        }
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

@router.get("/{feedback_id}")
async def get_feedback_by_id(feedback_id: int, db: Session = Depends(get_db)):
    try:
        feedback = db.query(Feedback).filter(Feedback.id == feedback_id).first()
        if not feedback:
            raise HTTPException(status_code=404, detail="反馈不存在")
        return {
            "id": feedback.id,
            "content": feedback.content,
            "type": feedback.type,
            "rating": feedback.rating,
            "sentiment": feedback.sentiment,
            "sentiment_score": feedback.sentiment_score,
            "status": feedback.status,
            "priority": feedback.priority,
            "player_name": feedback.player_name,
            "created_at": feedback.created_at.isoformat(),
            "updated_at": feedback.updated_at.isoformat()
        }
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

@router.put("/{feedback_id}")
async def update_feedback(feedback_id: int, update: FeedbackUpdate, db: Session = Depends(get_db)):
    try:
        feedback = db.query(Feedback).filter(Feedback.id == feedback_id).first()
        if not feedback:
            raise HTTPException(status_code=404, detail="反馈不存在")
        
        if update.status:
            feedback.status = update.status
        if update.priority:
            feedback.priority = update.priority
        
        db.commit()
        db.refresh(feedback)
        return {"message": "反馈更新成功"}
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

@router.delete("/{feedback_id}")
async def delete_feedback(feedback_id: int, db: Session = Depends(get_db)):
    try:
        feedback = db.query(Feedback).filter(Feedback.id == feedback_id).first()
        if not feedback:
            raise HTTPException(status_code=404, detail="反馈不存在")
        
        db.delete(feedback)
        db.commit()
        return {"message": "反馈删除成功"}
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))